x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/ReplaceStatsFilteredAggWithEvalTests.java

```diff
@@ -0,0 +1,736 @@
+        var leftLimit = as(left.child(), Limit.class);
+        as(leftLimit.child(), EsRelation.class);
+
+        var right = as(ij.right(), Aggregate.class);
+        assertThat(Expressions.names(right.aggregates()), contains("m1"));
+        assertEquals(1, right.aggregates().size());
+        assertTrue(right.aggregates().get(0).children().get(0) instanceof Count);
+        assertEquals("true", ((Count) right.aggregates().get(0).children().get(0)).filter().toString());
+
+        var source = as(right.child(), StubRelation.class);
+        assertThat(Expressions.names(source.output()), contains("emp_no", "salary", "gender"));
+    }
+
+    /**
+     * EsqlProject[[salary{f}#10, count(salary) where false{r}#4]]
+     * \_Eval[[0[LONG] AS count(salary) where false#4]]
+     *   \_Limit[1000[INTEGER],false]
+     *     \_EsRelation[test][_meta_field{f}#11, emp_no{f}#5, first_name{f}#6, ge..]
+     */
+    public void testReplaceInlinestatsFilteredAggWithEvalCount() {
+        var plan = plan("""
+            from test
+            | keep salary
+            | inlinestats count(salary) where false
+            """);
+        var project = as(plan, EsqlProject.class);
+        assertThat(Expressions.names(project.projections()), contains("salary", "count(salary) where false"));
+
+        var eval = as(project.child(), Eval.class);
+        assertThat(eval.fields().size(), is(1));
+        var alias = as(eval.fields().getFirst(), Alias.class);
+        assertThat(alias.name(), is("count(salary) where false"));
+        assertTrue(alias.child().foldable());
+        assertThat(alias.child().fold(FoldContext.small()), is(0L));
+
+        var limit = as(eval.child(), Limit.class);
+        as(limit.child(), EsRelation.class);
+    }
+
+    /**
+     * EsqlProject[[salary{f}#10, count_distinct(salary   2)   3 where false{r}#4]]
+     * \_Eval[[3[LONG] AS count_distinct(salary   2)   3 where false#4]]
+     *   \_Limit[1000[INTEGER],false]
+     *     \_EsRelation[test][_meta_field{f}#11, emp_no{f}#5, first_name{f}#6, ge..]
+     */
+    public void testReplaceInlinestatsFilteredAggWithEvalCountDistinctInExpression() {
+        var plan = plan("""
+            from test
+            | keep salary
+            | inlinestats count_distinct(salary + 2) + 3 where false
+            """);
+        var project = as(plan, EsqlProject.class);
+        assertThat(Expressions.names(project.projections()), contains("salary", "count_distinct(salary + 2) + 3 where false"));
+
+        var eval = as(project.child(), Eval.class);
+        assertThat(eval.fields().size(), is(1));
+        var alias = as(eval.fields().getFirst(), Alias.class);
+        assertThat(alias.name(), is("count_distinct(salary + 2) + 3 where false"));
+        assertTrue(alias.child().foldable());
+        assertThat(alias.child().fold(FoldContext.small()), is(3L));
+
+        var limit = as(eval.child(), Limit.class);
+        as(limit.child(), EsRelation.class);
+    }
+
+    /**
```

Comment:
```suggestion
    /*
```
to prevent having the block be interpreted as javadoc. Here and above, where there's no @code wrapping (which isn't ideal, as it's wrapped in IntelliJ; but that was just copied).